x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/utils.ts

```diff
@@ -193,6 +195,116 @@ export const getTupleDuplicateErrorsAndUniqueRules = (
   return [Array.from(errors.values()), Array.from(rulesAcc.values())];
 };
 
+// functions copied from here
+// https://github.com/elastic/kibana/blob/4584a8b570402aa07832cf3e5b520e5d2cfa7166/src/core/server/saved_objects/import/lib/check_origin_conflicts.ts#L55-L57
+const createQueryTerm = (input: string) => input.replace(/\\/g, '\\\\').replace(/\"/g, '\\"');
+const createQuery = (type: string, id: string, rawIdPrefix: string) =>
+  `"${createQueryTerm(`${rawIdPrefix}${type}:${id}`)}" | "${createQueryTerm(id)}"`;
+
+/**
+ * Query for a saved object with a given origin id and replace the
+ * id in the provided action with the _id from the query result
+ * @param action
+ * @param esClient
+ * @returns
+ */
+export const swapActionIds = async (
+  action: Action,
+  savedObjectsClient: SavedObjectsClientContract
+): Promise<Action | Error> => {
+  try {
+    // TODO: might need to pass down namespace
+    // but for now it's blank to do local testing.
+    const search = createQuery('action', action.id, '');
+    const foundAction = await savedObjectsClient.find<Action>({
+      type: 'action',
+      search,
+      rootSearchFields: ['_id', 'originId'],
+    });
+
+    if (foundAction.saved_objects.length === 1) {
+      return { ...action, id: foundAction.saved_objects[0].id };
+    } else if (foundAction.saved_objects.length > 1) {
+      return new Error(
+        `action connector with originId: ${action.id} had conflicts. Please resolve these conflicts either in the file you are attempting to upload or resolve the conflicting action connector in Kibana.`
```

Comment:
Should we be more specific about the conflict? Specify a conflicting id found? This might also be a bit confusing given the option we give of overwriting when conflicting ids found (could be a quick fix of adding context in the docs about this scenario).

Reply:
Good call. What do you think of replacing the error message with the following?

```
Found two action connectors with originId or _id: ${action.id} The upload cannot be completed unless the _id or the originId of the action connector is changed. See https://www.elastic.co/guide/en/kibana/current/sharing-saved-objects.html for more details
```

Reply:
I think that'd do the trick!